pkg/controller/servicebindingrequest/reconciler_test.go

```diff
@@ -212,22 +240,15 @@ func TestReconcilerGenericBinding(t *testing.T) {
 
 	// Reconcile without deployment
 	res, err := reconciler.Reconcile(reconcileRequest())
-	require.NoError(t, err)
+	require.EqualError(t, err, ErrApplicationNotFound.Error())
```

Comment:
Are we requeing when the application is not found, i.e application was there but then deleted, i.e I initially had an applicationSelector but now my deployment was deleted. If that happens then we will keep falling in requeue and this error will exist https://github.com/redhat-developer/service-binding-operator/issues/419

Reply:
I will look into this

Reply:
@Avni-Sharma review 7889bfa